TCD1304.py

```diff
@@ -77,11 +73,9 @@ def analog_signal_read (self, block:False):
         self.scope.select_range('CH1', max_voltage_output)   # voltagerange should be fitted to the sensors output for better resolution. sensor otput is between 2V-3V due to datasheet
         self.scope.configure_trigger(channel = 'CH1', voltage = min_voltage_output ) # starts recording , when voltage is over a certain level
         self.scope.capture(channels=1, samples=integration_elements*samples_per_element, timegap=integration_time/samples_per_element, block=False)
-        self.icg_clock(self)
-        x, y= self.scope.fetch_data()
-        diff = abs(x,y[1, 0] - min_voltage_output )
-        analog_measurement = np.ndarray ()
-        return analog_measurement  # this should put the Measurements and timestamps in an array
+        icg_clock()

```

Comment:
You still need to access the method via `self`: `self.icg_clock()`

Reply:
> You should not specify self as an explicit argument to method calls. self is only used in assignments (self.something = ...) and use of the assigned object, and method definitions (def mymethod(self, ...):).

I am quite confused...
So `self.icg_clock()` is the way to go?

Reply:
Yes.